### PR TITLE
Role binding for serviceaccount across namesapce.

### LINKED ISSUE #2651: RoleBinding subject name not transformed to match new ServiceAccount name
Hi, it seems that Kustomize isn't correctly replacing a RoleBinding `subjects.name` field when the ServiceAccount it references undergoes a name transformation (using `nameSuffix`). 

I'm not sure if this is expected behavior or not.

```console
$ kustomize version
{Version:kustomize/v3.5.5 GitCommit:897e7b6e61e65188d846c32bd3af9ef68b0f746a BuildDate:2020-05-11T16:51:33Z GoOs:linux GoArch:amd64}
```

To reproduce:

```bash
mkdir -p test/base
mkdir -p test/overlay

# Base, in namespace ns1
cat <<EOF >test/base/kustomization.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- sa.yaml
EOF

cat <<EOF >test/base/sa.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: my-sa
  namespace: ns1
EOF

# Overlay, in namespace ns2, but referencing sa in ns1
cat <<EOF >test/overlay/kustomization.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- ../base
- rbac.yaml
nameSuffix: -ns2
EOF

cat <<EOF >test/overlay/rbac.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: my-role
  namespace: ns2
rules:
  - apiGroups:
      - '*'
    resources:
      - '*'
    verbs:
      - '*'
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: my-role-binding
  namespace: ns2
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: my-role
subjects:
  - kind: ServiceAccount
    name: my-sa
    namespace: ns1
EOF
```

Output when running `kustomize build test/overlay`:
```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: my-sa-ns2
  namespace: ns1
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: my-role-ns2
  namespace: ns2
rules:
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: my-role-binding-ns2
  namespace: ns2
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: my-role-ns2
subjects:
- kind: ServiceAccount
  name: my-sa
  namespace: ns1
```

Desired output when running `kustomize build test/overlay` (RoleBinding's `subject.name` should be updated to match SA's new name):
```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: my-sa-ns2
  namespace: ns1
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: my-role-ns2
  namespace: ns2
rules:
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: my-role-binding-ns2
  namespace: ns2
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: my-role-ns2
subjects:
- kind: ServiceAccount
  name: my-sa-ns2
  namespace: ns1
```

Is this expected behavior? If yes, how can I use Kustomize to maintain a RoleBinding which references a SA in a separate base, while still allowing a name change of the SA from the overlay?

### PR DESCRIPTION
**Pull request by Shell32-Natsu**
Fixes #2651

### CODE REVIEW

**Comment by chrisob** [comment]
Thanks for this, you beat me to creating this test :grin:

**Comment by monopole** [comment]
So this looks great, but a small favor.

The first commit looks like it creates the test that establishes existing behavior, but github doesn't note that (run all tests and maintain the record).

Can we check that in first in a distinct PR?

Then this PR adds the change in behavior and change in test.  Thanks!

File: api/resmap/resmap.go [hunk 1]
```diff
@@ -611,19 +611,44 @@ func (m *resWrangler) SubsetThatCouldBeReferencedByResource(
...
+	subjectNamespaces := getNamespacesForRoleBinding(inputRes)
 	for _, r := range m.Resources() {
 		// Need to match more accuratly both at the time of selection and transformation.
 		// OutmostPrefixSuffixEquals is not accurate enough since it is only using
 		// the outer most suffix and the last prefix. Use PrefixedSuffixesEquals instead.
 		resId := r.CurId()
-		if (!isInputIdNamespaceable || !resId.IsNamespaceableKind() || resId.IsNsEquals(inputId)) &&
-			r.InSameKustomizeCtx(rctxm) {
+		if (!isInputIdNamespaceable || !resId.IsNamespaceableKind() || resId.IsNsEquals(inputId) ||
+			subjectNamespaces[r.GetNamespace()]) && r.InSameKustomizeCtx(rctxm) {
```
Comment (monopole) [inline api/resmap/resmap.go:621]: Please factor this out to a boolean method with a name summarizing what it does!

**Pull request merged**